File: 30 days of code/__ Java/Day 18 - Queues and stacks.java
/**
 *              QUEUES AND STACKS | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 18
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        String input = scan.next();
        scan.close();

        QueueStack qs = new QueueStack();
        char[] word = input.toCharArray();

        for (char c : word) {
            qs.pushCharacter(c);
            qs.enqueueCharacter(c);
        }

        boolean isPalindrome = true;
        for (int i = 0; i < word.length / 2; ++i) {
            if (qs.popCharacter() != qs.dequeueCharacter()) {
                isPalindrome = false;
                break;
            }
        }

        System.out.println("The word, " + input + ", is "
            + ((isPalindrome) ? "a palindrome." : "not a palindrome."));
    }
}

class QueueStack {
    private Stack<Character> stack;             // FILO
    private Queue<Character> queue;             // FIFO

    QueueStack() {
        stack = new Stack<>();
        queue = new LinkedList<>();
    }

    // Push char to STACK - current char is the first element
    void pushCharacter(char c) {
        stack.push(c);
    }

    // Add char to QUEUE - current char is the last element
    void enqueueCharacter(char c) {
        queue.add(c);
    }

    // Get char, from last to first
    char popCharacter() {
        return stack.pop();
    }

    // Get char, from first to last
    char dequeueCharacter() {
        return queue.poll();
    }
}

File: 30 days of code/__ C++/Day 18 - Queues and stacks.cpp
/**
*              QUEUES AND STACKS | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 18
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <string>
#include <stack>
#include <queue>

using namespace std;

ifstream file("input.txt");
istream& f = file;

class QueueStack {
public:
	void pushCharacter(char c) {
		_stack.push(c);
	}

	void enqueueCharacter(char c) {
		_queue.push(c);
	}

	char popCharacter() {
		char c = _stack.top();
		_stack.pop();
		return c;
	}

	char dequeueCharacter() {
		char c = _queue.front();
		_queue.pop();
		return c;
	}

private:
	stack<char> _stack;
	queue<char> _queue;
};

int main() {
	string input; f >> input;

	QueueStack qs;
	const char* word = input.c_str();

	int i = 0;
	while (word[i] != '\0') {
		qs.enqueueCharacter(word[i]);
		qs.pushCharacter(word[i]);
		i++;
	}

	bool isPalindrome = true;
	for (int i = 0; i < input.length(); ++i) {
		if (qs.popCharacter() != qs.dequeueCharacter()) {
			isPalindrome = false;
			break;
		}
	}

	cout << "The word, " << input << ", is " <<
		((isPalindrome) ? "a palindrome." : "not a palindrome.") << endl;

	return 0;
}